command/agent/dns.go

```diff
@@ -625,6 +627,7 @@ func (d *DNSServer) handleRecurse(resp dns.ResponseWriter, req *dns.Msg) {
 	var err error
 	for _, recursor := range d.recursors {
 		r, rtt, err = c.Exchange(req, recursor)
+		r.Compress = true
```

Comment:
This should probably be moved after the error check, because `r` may be nil in that case and cause a nil pointer dereference.